### PR TITLE
[webpack] Optimize startup time.

### PR DESCRIPTION
**Pull request by alloy**
~~⚠️ This depends on https://github.com/artsy/reaction/pull/1277, which is also why the tests currently fail.~~

* Replaces the progress plugin with one that also offers helpful `verbose` output.
* The separation of server/client code reduces the total amount of modules being considered by webpack by ~1000.
* Reduces startup time with a hot cache from ~70s to ~8s 🔥

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -1,83 +1,108 @@
...
     // Remove moment.js localization files
     new webpack.IgnorePlugin(/^\.\/locale$/, /moment$/),
-
+    new HardSourceWebpackPlugin({
+      cacheDirectory: path.join(cacheDirectory, "hard-source"),
+      info: {
+        mode: "none",
+        level: "error",
+      },
+    }),
```
Comment (damassi) [inline webpack.config.js]: I've tried this plugin before but noticed that it seemed to slow things down a bit, as opposed to speeding things up -- was probably misconfigured in some way.

File: src/desktop/apps/artist2/server.js [hunk 2]
```diff
@@ -1,15 +1,15 @@
-import { buildServerApp } from 'reaction/Artsy/Router'
-import { Meta, query, toJSONLD } from './components/Meta'
-import { renderLayout } from '@artsy/stitch'
-import { routes } from 'reaction/Apps/Artist/routes'
-import express from 'express'
-import metaphysics from 'lib/metaphysics.coffee'
-import React from 'react'
-import styled from 'styled-components'
+import { buildServerApp } from "reaction/Artsy/Router/server"
```
Comment (damassi) [inline src/desktop/apps/artist2/server.js]: @alloy - so this was the culprit? Makes sense (oversight)

File: webpack.config.js (hunk 1 above)
Comment (alloy) [reply webpack.config.js]: The initial definitely takes a bit longer but subsequent builds are like night and day.

File: src/desktop/apps/artist2/server.js (hunk 2 above)
Comment (alloy) [reply src/desktop/apps/artist2/server.js]: Yup, this was it!

File: src/desktop/config.coffee [hunk 3]
```diff
@@ -79,7 +79,7 @@ module.exports =
   METAPHYSICS_ENDPOINT: null
   MOBILE_MEDIA_QUERY: "only screen and (max-width: 640px)"
   MOBILE_URL: 'https://m.artsy.net'
-  NEW_ARTIST_PAGE_THRESHOLD: 0 # by default/not set no users will get the new page.
+  NEW_ARTIST_PAGE_THRESHOLD: 10 # by default/not set no users will get the new page.
```
Comment (alloy) [inline src/desktop/config.coffee]: @mzikherman cc

**Comment by alloy** [comment]
Merge on green

**Comment by mzikherman** [comment]
Can we discuss possibly reverting this?

It seems to make Force startups take a lot longer. Our dev flow (linking to Reaction, unlinking to published versions, toggling ENV vars for BNMO and AB tests and the like, etc.) require restarting Force often. Additionally, occasional `yarn` faffing and annoyances increase the likelihood of needing to restart Force extra times.

Furthermore, I just spent a long time debugging a mysterious 'dependency not found' issue, and did every `yarn`-related thing I could think of, only to have to be forced to do a `rm -rf .cache/`, which finally fixed the issue, only to then incur a penalty on startup.

I think others, anecdotally, have been discussing Force actually seeming slower than before.

I love the enthusiasm and idea of this PR, but I think we've seen more problems creep in than this was possibly trying to solve.

**Pull request merged**